flake.nix

```diff
@@ -2,63 +2,117 @@
   inputs = {
     flake-parts.url = "github:hercules-ci/flake-parts";
     nixpkgs.url = "github:NixOS/nixpkgs/nixos-24.11";
-    lynx.url = "github:the-computer-club/lynx";
+    lynx.url = "github:the-computer-club/lynx/flake-guard-v2";
     asluni.url = "github:the-computer-club/automous-zones";
     git-hooks-nix.url = "github:cachix/git-hooks.nix";
-
+    sops.url = "github:Mic92/sops-nix";
     disko = {
       url = "github:nix-community/disko";
       inputs.nixpkgs.follows = "nixpkgs";
     };
 
     # Sketti config, for user packages
-    terra.url = "github:skettisouls/nixos";
+    terra.url = "github:skettisouls/nixos/ad97f1a34696466582d0cc79ab64ee8a39b89294";
   };
 
   outputs = inputs @ { self, nixpkgs, flake-parts, ... }:
     flake-parts.lib.mkFlake { inherit inputs; }
-      ({ ... }:
+      (args@{ config, lib, ... }:
         let
-          inherit (nixpkgs) lib;
-          fileAttrs = lib.filterAttrs (_: t: t == "regular") (builtins.readDir ./src);
-          fileList = with builtins; filter (lib.hasSuffix "nix") (attrNames fileAttrs);
-          src = map (file: ./src/${file}) fileList;
+          inherit (nixpkgs.lib) nixosSystem;
+          rootConfig = config;
+          this = (import ./lib.nix args);
+          src = map (file: ./src/${file})
+            (this.walkNixFiles ./src);
         in
         {
           imports = with inputs; [
-            lynx.flakeModules.builtins
-            lynx.flakeModules.flake-guard
-            asluni.flakeModules.asluni
             git-hooks-nix.flakeModule
+            lynx.flakeModules.flake-guard
+            ./check-users.nix
           ];
 
           config = {
             systems = [ "x86_64-linux" ];
+            flake = {
+              ##################
+              # intended for
+              # note: `nix repl`
+              lib = lib.fix (
+                lib.extends
+                  (f: p: this)
+                  (f: lib)
+              );
+              ##################
+
+              modules.nixos = with inputs; [
+                disko.nixosModules.disko
+                lynx.nixosModules.flake-guard-host
+                asluni.nixosModules.asluni
+                sops.nixosModules.sops
+                { environment.etc.nixos.source = self; }
+                { environment.etc.nixpkgs.source = nixpkgs; }
```

Comment:
can we add a comment explaining why we have these here?

Reply:
These reference the source derivations used to build the code. symlinks are added to /etc in such that you can point other tooling to it under scrutiny. eg, debugging